### PR TITLE
Add External Texture VideoFrame visibileRect Tests

### PR DESCRIPTION
**Pull request by bjjones**
Issue: #910

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/web_platform/external_texture/video.spec.ts [hunk 1]
```diff
@@ -339,6 +338,112 @@ it will honor rotation metadata.
...
+      // All tested videos are derived from an image showing yellow, red, blue or green in each
+      // quadrant. In this test we crop the video to each quadrant and check that desired color
+      // is sampled from each corner of the cropped image.
+      const srcVideoHeight = 240;
+      const srcVideoWidth = 320;
+      let cropParams = [
+        // Top left (yellow)
+        {
+          _subRect: { x: 0, y: 0, width: srcVideoWidth / 2, height: srcVideoHeight / 2 },
+          _color: t.params._yellowExpectation,
```
Comment (austinEng) [inline src/webgpu/web_platform/external_texture/video.spec.ts:368]: these probably don't need underscores in key. just `subRect` and `color` is OK

**Review by austinEng — APPROVED** [review]
do the tests pass in chromium?

**Comment by bjjones** [comment]
> do the tests pass in chromium?

@austinEng Yes, the non-zero copy ones do, anyway. Not sure that I need that param right now.

I'm failing checks because I'm creating a VideoFrame from an overload that takes a VideoFrame. I'm not sure this why I'm seeing the error with the npm test. Is it possible the overload I'm using isn't spec'd? or am I using it wrong? I got the idea from [here](https://github.com/w3c/webcodecs/issues/281). This does work without error in Chromium.

File: src/webgpu/web_platform/external_texture/video.spec.ts (hunk 1 above)
Comment (bjjones) [reply src/webgpu/web_platform/external_texture/video.spec.ts]: Done.

**Comment by austinEng** [comment]
seems fine to have the zero copy param if it's expected to eventually be zero copy.

VideoFrame should be part of CanvasImageSource in the latest types https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L19212

we might need to roll the types, but if you'd like you can suppress the warning for now with a ts-ignore comment

**Comment by kainino0x** [comment]
> VideoFrame should be part of CanvasImageSource in the latest types https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L19212

Unfortunately this seems to have been added only 4 days ago, and hasn't made it into a TypeScript release.
https://github.com/microsoft/TypeScript/commit/bb566f997720926597fc0f680c8c5a24fd47f302

Just use some typecasts to make it work for now, leave a MAINTENANCE_TODO to remove the typecasts with TypeScript 4.9.6.

**Comment by kainino0x** [comment]
Pushed a commit to add the cast, you can merge if ready

**Comment by bjjones** [comment]
Thank you for finishing this up, Kai.

**Pull request merged**